cmd/influx/task.go

```diff
@@ -3,342 +3,425 @@ package main
 	}
-	if taskCreateFlags.org.id != "" || taskCreateFlags.org.name != "" {
-		svc, err := newOrganizationService()
-		if err != nil {
-			return nil
-		}
-		oid, err := taskCreateFlags.org.getID(svc)
+	if t.org.id != "" || t.org.name != "" {
+		oid, err := t.org.getID(orgSvc)
 		if err != nil {
 			return fmt.Errorf("error parsing organization ID: %s", err)
 		}
 		tc.OrganizationID = oid
 	}
 
-	t, err := s.CreateTask(context.Background(), tc)
+	tsk, err := tskSvc.CreateTask(context.Background(), tc)
 	if err != nil {
 		return err
 	}
 
-	return printTasks(
-		cmd.OutOrStdout(),
-		taskPrintOpts{
-			hideHeaders: taskPrintFlags.hideHeaders,
-			json:        taskPrintFlags.json,
-			task:        t,
-		},
-	)
+	return t.printTasks(taskPrintOpts{task: tsk})
 }
 
-var taskFindFlags struct {
+type taskFindFlags struct {
 	user    string
-	id      string
 	limit   int
 	headers bool
-	org     organization
 }
 
-func taskFindCmd(f *globalFlags, opt genericCLIOpts) *cobra.Command {
-	cmd := opt.newCmd("list", taskFindF, true)
+func (t *cmdTaskBuilder) taskFindCmd() *cobra.Command {
+	cmd := t.opts.newCmd("list", t.taskFindF, true)
 	cmd.Short = "List tasks"
 	cmd.Aliases = []string{"find", "ls"}
 
-	taskFindFlags.org.register(opt.viper, cmd, false)
-	f.registerFlags(opt.viper, cmd)
-	registerPrintOptions(opt.viper, cmd, &taskPrintFlags.hideHeaders, &taskPrintFlags.json)
-	cmd.Flags().StringVarP(&taskFindFlags.id, "id", "i", "", "task ID")
-	cmd.Flags().StringVarP(&taskFindFlags.user, "user-id", "n", "", "task owner ID")
-	cmd.Flags().IntVarP(&taskFindFlags.limit, "limit", "", influxdb.TaskDefaultPageSize, "the number of tasks to find")
-	cmd.Flags().BoolVar(&taskFindFlags.headers, "headers", true, "To print the table headers; defaults true")
+	t.org.register(t.opts.viper, cmd, false)
+	t.globalFlags.registerFlags(t.opts.viper, cmd)
+	registerPrintOptions(t.opts.viper, cmd, &t.taskPrintFlags.hideHeaders, &t.taskPrintFlags.json)
+	cmd.Flags().StringVarP(&t.taskID, "id", "i", "", "task ID")
+	cmd.Flags().StringVarP(&t.taskFindFlags.user, "user-id", "n", "", "task owner ID")
+	cmd.Flags().IntVarP(&t.taskFindFlags.limit, "limit", "", influxdb.TaskDefaultPageSize, "the number of tasks to find")
+	cmd.Flags().BoolVar(&t.taskFindFlags.headers, "headers", true, "To print the table headers; defaults true")
 
 	return cmd
 }
 
-func taskFindF(cmd *cobra.Command, args []string) error {
-	if err := taskFindFlags.org.validOrgFlags(&flags); err != nil {
+func (t *cmdTaskBuilder) taskFindF(cmd *cobra.Command, args []string) error {
+
+	if err := t.org.validOrgFlags(&flags); err != nil {
 		return err
 	}
 
-	client, err := newHTTPClient()
+	tskSvc, _, err := t.svcFn()
 	if err != nil {
 		return err
 	}
 
-	s := &http.TaskService{
-		Client: client,
-	}
-
 	filter := influxdb.TaskFilter{}
-	if taskFindFlags.user != "" {
-		id, err := influxdb.IDFromString(taskFindFlags.user)
+	if t.taskFindFlags.user != "" {
+		id, err := influxdb.IDFromString(t.taskFindFlags.user)
```

Comment:
Unfortunate that `--user` expects and ID while `--org` expects a name, but I guess that's existing behavior.

Reply:
Big fax.. I think consistency can be accomplished if one were to edit `influxdb.TaskFilter`'s Organization field to ID (and similarly edit the functions that use the field)